protocols/luci-proto-bonding/htdocs/luci-static/resources/protocol/bonding.js

```diff
@@ -0,0 +1,442 @@
+'use strict';
+'require ui';
+'require uci';
+'require form';
+'require fs';
+'require network';
+'require tools.widgets as widgets';
+
+
+function getSelectableSlaves(section_id) {
+	var rv = [];
+	var NonUsableMac = /^(00:00:00:00:00:00|null)/;
+	var interfaces = uci.sections('network', 'interface');
+
+	return network.getDevices().then(function(devices) {
+		for (var i = 0; i < devices.length; i++) {
+			var in_use = false;
+			var NotUsable = NonUsableMac.test(devices[i].getMAC());
+
+			// Only "real" interfaces for slaves needed
+			if (NotUsable == false) {
+				for (var j = 0; j < interfaces.length; j++) {
+					if (uci.get('network', interfaces[j]['.name'], 'proto') == 'bonding') {
+						var slaves = uci.get('network', interfaces[j]['.name'], 'slaves');
```

Comment:
I suggest the following here:
`var slaves = L.toArray(uci.get('network', interfaces[j]['.name'], 'slaves'));`

The has a few benefits:
 - UCI lists are retained as-is
 - UCI options are converted to lists by breaking the value on spaces
 - Empty/unset values are converted to an empty list so you do not need an extra `if (slaves != null)` check

Reply:
OK, I made it following your suggestion.